Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -2243,44 +2244,78 @@ extension BrowserViewController: UIAdaptivePresentationControllerDelegate {
+            presentRebrandIntro()
+        } else {
+            presentInsightfulSheetsIfNeeded()
         }
-        presentDefaultBrowserPromoIfNeeded()
+    }
+
+    private func presentLoadingScreen() {
+        present(LoadingScreen(profile: profile, referrals: referrals, referralCode: User.shared.referrals.pendingClaim), animated: true)
+    }
+
+    private func handleFirstTimeUserActions() {
+        User.shared.firstTime = false
+        User.shared.migrated = true
+        User.shared.hideRebrandIntro()
+        // deactivate searchbar hint for new users
+        contextHintVC.viewModel.markContextualHintPresented()
+    }
+
+    private func presentRebrandIntro() {
+        let intro = NTPIntroViewController()
+        intro.modalPresentationStyle = .overFullScreen
+        intro.modalTransitionStyle = .crossDissolve
+        present(intro, animated: true)
+        User.shared.hideRebrandIntro()
     }
 
     private func showLoadingScreen(for user: User) -> Bool {
         (user.migrated != true && !user.firstTime)
-            || user.referrals.pendingClaim != nil
+                || user.referrals.pendingClaim != nil
     }
 
-    func presentDefaultBrowserPromoIfNeeded() {
-        let isHome = tabManager.selectedTab?.url.flatMap { InternalURL($0)?.isAboutHomeURL } ?? false
+    func presentInsightfulSheetsIfNeeded() {
+        if isHomePage() && presentedViewController == nil && !showLoadingScreen(for: .shared) {
+            if !presentDefaultBrowserPromoIfNeeded() {
+                presentWhatsNewPageIfNeeded()
+            }
+        }
+    }
 
-        guard isHome,
-              presentedViewController == nil,
-              !showLoadingScreen(for: .shared),
-              !User.shared.showsRebrandIntro else { return }
+    private func isHomePage() -> Bool {
+        tabManager.selectedTab?.url.flatMap { InternalURL($0)?.isAboutHomeURL } ?? false
+    }
 
-        if shouldShowIntroScreen && DefaultBrowserExperiment.minPromoSearches() <= User.shared.treeCount  {
-            if #available(iOS 14, *) {
-                let defaultPromo = DefaultBrowser(delegate: self)
-                present(defaultPromo, animated: true)
-            } else {
-                profile.prefs.setInt(1, forKey: PrefsKeys.IntroSeen)
-            }
+    @discardableResult
+    private func presentWhatsNewPageIfNeeded() -> Bool {
+        guard !User.shared.showsRebrandIntro, shouldShowWhatsNewPageScreen else { return false }
+
+        let provider = WhatsNewLocalDataProvider()
+        let viewModel = WhatsNewViewModel(provider: provider)
+
+        if !viewModel.items.isEmpty  {
+            WhatsNewViewController.presentOn(self)
+            return true
         }
+        return false
+    }
+
+    @discardableResult
+    private func presentDefaultBrowserPromoIfNeeded() -> Bool {
+        guard !User.shared.showsRebrandIntro, shouldShowIntroScreen, DefaultBrowserExperiment.minPromoSearches() <= User.shared.treeCount else { return false }
```

Comment:
Same as above for `User.shared.showsRebrandIntro`.

Also, got a bit confused with the `shouldShowIntroScreen` naming, somehow it does not seem to be related to default browser to me 😅 I know this already existed, just wondering if it makes sense to switch for clarity 🤔

Reply:
Wanted to keep the changes as minimal as possible but I totally agree on naming 😝